airflow/utils/operator_resources.py

```diff
@@ -134,3 +142,37 @@ def __eq__(self, other):
 
     def __repr__(self):
         return str(self.__dict__)
+
+    def to_dict(self):
+        return {
+            'cpus': self.cpus.to_dict(),
+            'ram': self.ram.to_dict(),
+            'disk': self.disk.to_dict(),
+            'gpus': self.gpus.to_dict(),
+        }
+
+    def to_json(self):
+        return json.dumps(self.to_dict())
+
+    @classmethod
+    def is_operator_resources_json_string(cls, json_str: str):
```

Comment:
I don't have a suggestion for better name, but would love a shorter name if someone has a suggestion.